src/discriminant-calculator.c

```diff
@@ -0,0 +1,70 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+
+struct CalculationResult {
+	int result;
+	int isSuccess;
+	int x1;
+	int x2;
+	char message[100];
+};
+
+struct CalculationResult rootCalculator(int discriminant, int a, int b, int c) {
+        struct CalculationResult result;
+	
+	if (discriminant >= 0) {
+		result.x1 = (-b + sqrt(discriminant) / (2 * a));
+		result.x2 = (-b - sqrt(discriminant) / (2 * a));
+		result.isSuccess = 1;
+		snprintf(result.message, sizeof(result.message), "Real roots found");
+	} else {
+		result.isSuccess = 0;
+		snprintf(result.message, sizeof(result.message), "Complex roots found");
+	}
+
+	return result;
+}
+
+struct CalculationResult discriminantCalculator(int a, int b, int c) {
```

Comment:
Why this function calculates the roots?
The responsibility of this function is to calculate the `discriminant`

Reply:
done